modules/clairs/1.0/clairs.smk

```diff
@@ -0,0 +1,307 @@
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam.crai"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+        op.absolute_symlink(input.bai, output.crai)
+
+
+rule _clairs_input_normal_bam:
+    input:
+        bam = CFG["inputs"]["normal_bam"],
+        bai = CFG["inputs"]["normal_bai"],
+        normal_name = CFG["runs"]["normal_name"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_name}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_name}.bam.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_name}.bam.crai"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+        op.absolute_symlink(input.bai, output.crai)
+
+# Rule to see if current container version is most recent?
+
+rule _clairs_install:
+    output:
+        sif = CFG["dirs"]["clairs"] + "clairs.sif"
+    log:
+        stdout = CFG["logs"]["clairs"] + "install.stdout.log",
+        stderr = CFG["logs"]["clairs"] + "install.stderr.log"
+    conda:
+        CFG["conda_envs"]["singularity"]
+    shell:
+        """
+        singularity pull {output.sif} docker://hkubal/clairs:latest
+        """
+
+
+def get_platform(wildcards):
+    CFG = config["lcr-modules"]["clairs"]
+    this_sample = op.filter_samples(
+        CFG["runs"],
+        sample_id = wildcards.tumour_id,
+        seq_type = wildcards.seq_type, 
+        genome_build = wildcards.genome_build
+    )
+    platform = this_sample["tumour_platform"].tolist()[0]
+    return platform
+
+
+def get_normal(wildcards):
+    CFG = config["lcr-modules"]["clairs"]
+    this_sample = op.filter_samples(
+        CFG["runs"],
+        sample_id = wildcards.tumour_id,
+        seq_type = wildcards.seq_type, 
+        genome_build = wildcards.genome_build
+    )
+    normal = this_sample["normal_name"].tolist()[0]
+    return normal
+
+
+# Calls variants using ClairS
+rule _clairs_call_variants:
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
+        normal_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_name}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        fai = reference_files("genomes/{genome_build}/genome_fasta/genome.fa.fai"),
+        sif = str(rules._clairs_install.output.sif)
+    output:
+        vcf = temp(CFG["dirs"]["clairs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_name}--{chemistry}/output.vcf.gz"),
```

Comment:
normally you need to comress the vcf to generate the .gz format. In the shell command to this rule, there is no compression step. Does tslairs outputs compressed file by default - or does it make a file with .gz extension but it is not actually compressed?